packages/gatsby-recipes/src/cli/index.js

```diff
@@ -412,148 +534,58 @@ module.exports = async ({ recipe, graphqlPort, projectRoot }) => {
+        if (!sentContinue) {
+          sendEvent({ event: `CONTINUE` })
+          sentContinue = true
         }
-
-        return (
-          <Div>
-            <Div>
-              <Text bold italic marginBottom={2}>
-                Proposed changes
-              </Text>
-            </Div>
-            {state.context.plan.map((p, i) => (
-              <Div marginTop={1} key={`${p.resourceName} plan ${i}`}>
-                <Text italic>{p.resourceName}:</Text>
-                <Text> * {p.describe}</Text>
-                {p.diff && p.diff !== `` && (
-                  <>
-                    <Text>---</Text>
-                    <Text>{p.diff}</Text>
-                    <Text>---</Text>
-                  </>
-                )}
-              </Div>
-            ))}
-            <Div marginTop={1}>
-              <Color magentaBright>>> Press enter to run this step</Color>
-            </Div>
-          </Div>
-        )
       }
 
-      const RunningStep = ({ state }) => {
-        const isPlan = state.context.plan && state.context.plan.length > 0
-        const isRunningStep = state.value === `applyingPlan`
-
-        if (!isPlan || !isRunningStep) {
-          return null
-        }
+      // install mode
+      if (isInstallMode) {
+        isReady = state?.value === `applyingPlan` || state?.value === `done`
+      } else {
+        isReady = state?.value === `presentPlan`
+      }
 
+      if (!isReady) {
         return (
-          <Div>
-            {state.context.plan.map((p, i) => (
-              <Div key={`${p.resourceName}-${i}`}>
-                <Text italic>{p.resourceName}:</Text>
-                <Text>
-                  {` `}
-                  <Spinner /> {p.describe}
-                  {` `}
-                  {state.context.elapsed > 0 && (
-                    <Text>({state.context.elapsed / 1000}s elapsed)</Text>
-                  )}
-                </Text>
-              </Div>
-            ))}
-          </Div>
+          <Text>
+            <Spinner.default /> Loading recipe
+          </Text>
         )
       }
 
-      const Error = ({ state }) => {
-        log(`errors`, state)
-        if (state && state.context && state.context.error) {
-          return (
-            <Color red>{JSON.stringify(state.context.error, null, 2)}</Color>
-          )
-        }
-
-        return null
-      }
+      const isDone = state.value === `done`
 
+      // If we're done with an error, render out error (happens below)
+      // then exit.
       if (state.value === `doneError`) {
-        return <Error width="100%" state={state} />
-      }
-
-      const staticMessages = {}
-      for (let step = 0; step < state.context.currentStep; step++) {
-        staticMessages[step] = [
-          {
-            type: `mdx`,
-            key: `mdx-${step}`,
-            value: state.context.steps[step],
-          },
-        ]
+        process.nextTick(() => process.exit())
```

Comment:
Please print a message that this is an explicit `.exit()` or you'll be tearing your hair out later when debugging something that relates to this.

Also, I'd expect this to be a non-zero error because an error happened. This is effectively swallowing the return error status.